File: Pages/ResponsePage.cs
using Microsoft.Playwright;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Microsoft.Playwright.NUnit;
using InputFormTests.Models;

namespace InputFormTests.Pages
{
    public class ResponsePage
    {
        private readonly IPage _page;
        private readonly ILocator _firstNameResponse;
        private readonly ILocator _secondNameResponse;
        private readonly ILocator _ageResponse;
        private readonly ILocator _countryResponse;
        private readonly ILocator _notesResponse;
        public ResponsePage(IPage page)
        {
            _page = page;
            _firstNameResponse = _page.Locator("#firstname-value");
            _secondNameResponse = _page.Locator("#surname-value");
            _ageResponse = _page.Locator("#age-value");
            _countryResponse = _page.Locator("#country-value");
            _notesResponse = _page.Locator("#notes-value");
        }
        public async Task CheckResponse(Submission submission)
        {
            await CheckField(_firstNameResponse, submission.FirstName, "First name");
            await CheckField(_secondNameResponse, submission.SecondName, "Second name");
            await CheckField(_ageResponse, submission.Age.ToString(), "Age");
            await CheckField(_countryResponse, submission.Country, "Country");
            await CheckField(_notesResponse, submission.Notes, "Notes");
        }
        private async Task CheckField(ILocator fieldLocator, string expectedValue, string fieldName)
        {
            var actualValue = await fieldLocator.InnerTextAsync();
            if (actualValue != expectedValue)
                throw new Exception($"{fieldName} mismatch: expected '{expectedValue}', got '{actualValue}'.");
        }

    }
}

File: pages/ResponsePage.ts
import { Page, Locator, expect } from '@playwright/test';
import { Submission } from '../models/Submission';

export class ResponsePage {
  private page: Page;
  private firstNameResponse: Locator;
  private secondNameResponse: Locator;
  private ageResponse: Locator;
  private countryResponse: Locator;
  private notesResponse: Locator;
  private responseHeader: Locator;

  constructor(page: Page) {
    this.page = page;
    this.firstNameResponse = page.locator('#firstname-value');
    this.secondNameResponse = page.locator('#surname-value');
    this.ageResponse = page.locator('#age-value');
    this.countryResponse = page.locator('#country-value');
    this.notesResponse = page.locator('#notes-value');
    this.responseHeader = page.getByText('Input Validation Response');
  }

  async assertPageIsNotLoaded(): Promise<void> {
    const isVisible = await this.responseHeader.isVisible();
    expect(isVisible).toBeFalsy();
  }

  async checkResponse(submission: Submission): Promise<void> {
    await this.checkField(this.firstNameResponse, submission.firstName, 'First name');
    await this.checkField(this.secondNameResponse, submission.secondName, 'Second name');
    await this.checkField(this.ageResponse, submission.age?.toString() || '', 'Age');
    await this.checkField(this.countryResponse, submission.country, 'Country');
    await this.checkField(this.notesResponse, submission.notes, 'Notes');
  }

  private async checkField(fieldLocator: Locator, expectedValue: string, fieldName: string): Promise<void> {
    const actualValue = await fieldLocator.innerText();
    if (actualValue !== expectedValue) {
      throw new Error(`${fieldName} mismatch: expected '${expectedValue}', got '${actualValue}'.`);
    }
  }
}
